core/trino-main/src/main/java/io/trino/operator/aggregation/partial/PartialAggregationController.java

```diff
@@ -29,16 +32,23 @@
  */
 public class PartialAggregationController
 {
-    private final long minNumberOfRowsProcessed;
+    /**
+     * Process enough pages to fill up partial-aggregation buffer few times before
+     * considering partial-aggregation to be turned off.
+     */
+    private static final int DISABLE_AGGREGATION_BUFFER_SIZE_TO_INPUT_BYTES_FACTOR = 2;
```

Comment:
I would say 2 is already conservative (e.g. to average between multiple buffer fills). Only required columns should be part of input. Since PA aggregation state is usually bigger than input columns, then it should fill up even faster with input pages where row is unique